django/template/base.py

```diff
@@ -934,17 +964,28 @@ def render(self, context):
 
     def render_annotated(self, context):
         """
-        Render the node. If debug is True and an exception occurs during
-        rendering, the exception is annotated with contextual line information
-        where it occurred in the template. For internal usage this method is
-        preferred over using the render method directly.
+        Render the node with debug information if an error occurs. For
+        internal usage this method is preferred over using the render method
+        directly.
         """
-        try:
+        with annotate_exception(context, self.token):
             return self.render(context)
-        except Exception as e:
-            if context.template.engine.debug and not hasattr(e, 'template_debug'):
-                e.template_debug = context.template.get_exception_info(e, self.token)
-            raise
+
+    def stream(self, context):
+        """
+        Return a string generator rendering the node.
+        """
+        yield self.render(context)
+
+    def stream_annotated(self, context):
+        """
+        Stream the node with debug information if an error occurs. For
+        internal usage this method is preferred over using the render method
+        directly.
+        """
+        with annotate_exception(context, self.token):
+            for chunk in self.stream(context):
+                yield chunk
```

Comment:
Would this yield every symbol of the string? Could this be the reason for slow performance?